lib/Shippo/Util.php

```diff
@@ -32,7 +32,7 @@ public static function convertShippoObjectToArray($values)
         $results = array();
         foreach ($values as $k => $v) {
             // FIXME: this is an encapsulation violation
-            if ($k[0] == '_') {
+            if (is_string($k) && $k[0] == '_') {
```

Comment:
This should be is_array($k) && ...
```suggestion
            if (is_array($k) && $k[0] == '_') {
```

Reply:
They’re using array access to retrieve the first character of the key. It’s not possible to key an array with another array.

Reply:
Oh got it now, your are correct, thank you!